### ISSUE TITLE
Failed upgrade from 2.4.0 to 2.5.1 - Error: FOREIGN KEY constraint failed

### ISSUE THREAD

**Issue by jdstrand** [influxdata/influxdb#23867]
When upgrading from 2.4.0 to 2.5.1 via apt, influxdb didn't come up. I don't have a general reproducer, but I downgraded influxdb back to 2.4.0 via apt and everything is working again (and presume if I tried to upgrade it would fail again). Relevant details:

* Ubuntu 18.04 LTS amd64 host, fully up to date (`Linux 4.15.0-196-generic x86_64`)
* influxdb2 2.4.0 deb is installed `InfluxDB v2.4.0 (git: de247bab08) build_date: 2022-08-18T19:41:15Z`:
    ```
    $ apt-cache policy influxdb2
    influxdb2:
      Installed: 2.4.0-1
      Candidate: 2.5.1-1
      Version table:
         2.5.1-1 500
            500 https://repos.influxdata.com/ubuntu bionic/stable amd64 Packages
     *** 2.4.0-1 500
            500 https://repos.influxdata.com/ubuntu bionic/stable amd64 Packages
            100 /var/lib/dpkg/status
    ```
* I do have one replication setup, going to Cloud 2
* These are the files in `/var/lib/influxdata` after the failed upgrade and subsequent downgrade back to 2.4.0:
    ```
    $ sudo ls -l /var/lib/influxdb/
    total 3784
    drwxr-xr-x 5 influxdb influxdb    4096 Apr 15  2022 engine
    -rw------- 1 influxdb influxdb 1048576 Nov  6 18:26 influxd.bolt
    -rw-r--r-- 1 influxdb influxdb  815104 Jan  6  2022 influxd.bolt.pre-2.1.1-upgrade.backup
    -rw-r--r-- 1 influxdb influxdb  815104 Apr 15  2022 influxd.bolt.pre-v2.2.0-upgrade.backup
    -rw-r--r-- 1 influxdb influxdb  823296 Jun 22 09:19 influxd.bolt.pre-v2.3.0+SNAPSHOT.090f681737-upgrade.backup
    -rw-r--r-- 1 influxdb influxdb       6 Nov  5 10:56 influxd.pid
    -rw-r--r-- 1 influxdb influxdb  110592 Nov  2 10:35 influxd.sqlite
    -rw-r--r-- 1 influxdb influxdb   53248 Apr 15  2022 influxd.sqlite.pre-v2.2.0-upgrade.backup
    -rw-r--r-- 1 influxdb influxdb   90112 Sep  3 02:15 influxd.sqlite.pre-v2.4.0-upgrade.backup
    -rw-r--r-- 1 influxdb influxdb  110592 Nov  5 10:51 influxd.sqlite.pre-v2.5.1-upgrade.backup
    ```

__Config:__
Copy any non-default config values here or attach the full config as a gist or file.
```
$ cat /etc/influxdb/config.toml
bolt-path = "/var/lib/influxdb/influxd.bolt"
engine-path = "/var/lib/influxdb/engine"
tls-cert = "/etc/ssl/influxdb/fullchain.pem"
tls-key = "/etc/ssl/influxdb/privkey.pem"
pprof-disabled = true
```

__Logs:__
```
Nov  5 10:33:39 myhost influxd-systemd-start.sh[2800]: Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: ts=2022-11-05T15:33:40.045710Z lvl=info msg="Welcome to InfluxDB" log_id=0dz6exIW000 version=v2.5.1 commit=5b6fdbf05d build_date=2022-11-02T18:06:28Z log_level=info
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: ts=2022-11-05T15:33:40.047836Z lvl=info msg="Resources opened" log_id=0dz6exIW000 service=bolt path=/var/lib/influxdb/influxd.bolt
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: ts=2022-11-05T15:33:40.048619Z lvl=info msg="Resources opened" log_id=0dz6exIW000 service=sqlite path=/var/lib/influxdb/influxd.sqlite
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: ts=2022-11-05T15:33:40.055530Z lvl=info msg="Backing up pre-migration metadata" log_id=0dz6exIW000 service="SQL migrations" backup_path=/var/lib/influxdb/influxd.sqlite.pre-v2.5.1-upgrade.backup
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: ts=2022-11-05T15:33:40.060478Z lvl=info msg="Bringing up metadata migrations" log_id=0dz6exIW000 service="SQL migrations" migration_count=1
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: ts=2022-11-05T15:33:40.062361Z lvl=error msg="Failed to apply SQL migrations" log_id=0dz6exIW000 error="FOREIGN KEY constraint failed"
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: Error: FOREIGN KEY constraint failed
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: See 'influxd -h' for help
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: Command "print-config" is deprecated, use the influx-cli command server-config to display the configuration values from the running server
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: TLS cert and key found -- using https
Nov  5 10:33:40 myhost influxd-systemd-start.sh[2800]: InfluxDB API at https://localhost:8086/ready unavailable after 1 attempts...
Nov  5 10:33:41 myhost influxd-systemd-start.sh[2800]: InfluxDB API at https://localhost:8086/ready unavailable after 2 attempts...
Nov  5 10:33:42 myhost influxd-systemd-start.sh[2800]: InfluxDB API at https://localhost:8086/ready unavailable after 3 attempts...
Nov  5 10:33:43 myhost influxd-systemd-start.sh[2800]: InfluxDB API at https://localhost:8086/ready unavailable after 4 attempts...
Nov  5 10:33:44 myhost influxd-systemd-start.sh[2800]: InfluxDB API at https://localhost:8086/ready unavailable after 5 attempts...
...
```

I also have

**Comment by jeffreyssmith2nd** [comment]
> I downgraded influxdb back to 2.4.0

What were the steps you took to downgrade to 2.4.0?

**Comment by jeffreyssmith2nd** [comment]
Confirmed and replicated, looks like it is related to the replications migrations introduced in 2.5.

**Comment by jdstrand** [comment]
@jeffreyssmith2nd - curious if you see this fix as suitable for a 2.5.2 release or 2.6.0? (I ask cause at it is, I'm stuck on 2.4.0 which I'm fine with, but I mention since you may have more 2.4 -> 2.6 upgraders than is normal (depending on how many are using replications of course))
